backend/market/serializers.py

```diff
@@ -106,84 +112,121 @@ def contains_profanity(self, text):
         if (
-            self.context["request"].user == instance.subletter
+            self.context["request"].user == instance.seller
             or self.context["request"].user.is_superuser
         ):
             instance.delete()
         else:
-            raise serializers.ValidationError("You do not have permission to delete this sublet.")
+            raise serializers.ValidationError("You do not have permission to delete this item.")
 
 
-class SubletSerializerRead(serializers.ModelSerializer):
-    amenities = serializers.PrimaryKeyRelatedField(
-        many=True, queryset=Amenity.objects.all(), required=False
-    )
-    images = SubletImageURLSerializer(many=True, required=False)
+# simple tag serializer for use when pulling all serializers/etc
+class ItemSerializerSimple(serializers.ModelSerializer):
+    tags = serializers.PrimaryKeyRelatedField(many=True, queryset=Tag.objects.all(), required=False)
+    images = ItemImageURLSerializer(many=True, required=False)
+    category = serializers.PrimaryKeyRelatedField(queryset=Category.objects.all(), required=True)
+    seller = serializers.PrimaryKeyRelatedField(queryset=User.objects.all(), required=False)
 
     class Meta:
-        model = Sublet
-        read_only_fields = ["id", "created_at", "subletter", "sublettees"]
+        model = Item
         fields = [
             "id",
-            "subletter",
-            "amenities",
+            "seller",
+            "tags",
+            "category",
+            "favorites",
             "title",
-            "address",
-            "beds",
-            "baths",
-            "description",
-            "external_link",
             "price",
             "negotiable",
-            "start_date",
-            "end_date",
-            "expires_at",
             "images",
         ]
-
-
-# simple sublet serializer for use when pulling all serializers/etc
-class SubletSerializerSimple(serializers.ModelSerializer):
-    amenities = serializers.PrimaryKeyRelatedField(
-        many=True, queryset=Amenity.objects.all(), required=False
-    )
-    images = SubletImageURLSerializer(many=True, required=False)
-
-    class Meta:
-        model = Sublet
-        fields = [
+        read_only_fields = [
             "id",
-            "subletter",
-            "amenities",
+            "seller",
+            "tags",
+            "category",
+            "favorites",
             "title",
-            "address",
-            "beds",
-            "baths",
             "price",
             "negotiable",
-            "start_date",
-            "end_date",
             "images",
         ]
-        read_only_fields = ["id", "subletter"]
+
+
+class SubletSerializer(serializers.ModelSerializer):
+    item = ItemSerializer(required=True)
+
+    class Meta:
+        model = Sublet
+        fields = ["id", "item", "address", "beds", "baths", "start_date", "end_date"]
+        read_only_fields = ["id"]
+
+    def create(self, validated_data):
```

Comment:
Django doesn't support writable nested fields, so we need to explicitly state how to create Item.
However, I can call super.create to create the sublet object instead of creating it myself.

Reply:
gotcha, this looks good then!